rest-api/src/main/scala/za/co/absa/enceladus/rest_api/controllers/DatasetController.scala

```diff
@@ -113,7 +113,7 @@ class DatasetController @Autowired()(datasetService: DatasetService)
   def replaceProperties(@AuthenticationPrincipal principal: UserDetails,
                         @PathVariable datasetName: String,
                         @RequestBody newProperties: Optional[Map[String, String]]): CompletableFuture[ResponseEntity[Option[Dataset]]] = {
-    datasetService.replaceProperties(principal.getUsername, datasetName, newProperties.toScalaOption).map {
+    datasetService.updatePropertiesV2(principal.getUsername, datasetName, newProperties.toScalaOption).map {
```

Comment:
No, this has nothing to do with spring autowiring. This is just a way to have V2 and V3 implementation in some services so that the v2 version can be removed easily when only v3 is to stay.

If we want to keep v2 and v3 long-term, this would require to create separate services, e.g. `datasetService` and `datasetServiceV3`. One of these solutions is, I believe, necessary if you want to keep the original implementation working.

It then begs the question - what is the intended lifetime of V2 API? Knowing this better, it could help to design better: _copy-paste-and-change-a-few-things knowing that the V2 impl will be gone soon_ vs. keeping v3 extending v2 (no code duplication, but harder removal)

Reply:
TODO ... review again

Reply:
The v3 method moved to DatasetServiceV3, so now it is tidier. Thanks for the tip, @benedeki.